src/main/java/org/wildfly/security/auth/client/ConfigurationKeyManager.java

```diff
@@ -132,11 +132,20 @@ public PrivateKey getPrivateKey(final String alias) {
         return credential == null ? null : credential.getPrivateKey();
     }
 
+    /**
+     * Obtain a new {@link Builder} capable of building a {@link ConfigurationKeyManager}.
+     *
+     * @return a new {@link Builder} capable of building a {@link ConfigurationKeyManager}.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
     static final class Builder {
         private final Map<String, Map<X500Principal, String>> credentialAliasesByKeyTypeAndIssuer = new HashMap<>();
         private final Map<String, X509CertificateChainPrivateCredential> credentialsByAlias = new HashMap<>();
 
-        Builder() {
+        private Builder() {
```

Comment:
Package-private is better, because Java has to synthesize a package-private constructor anyway (inner classes are basically a hack in the JVM; they don't magically have the right rules for private).